Source/WebCore/rendering/svg/SVGTextMetricsBuilder.cpp

```diff
@@ -76,8 +76,19 @@ void SVGTextMetricsBuilder::advanceSimpleText()
 void SVGTextMetricsBuilder::advanceComplexText()
 {
     unsigned metricsLength = currentCharacterStartsSurrogatePair() ? 2 : 1;
```

Comment:
Do we need this? I think ComplexTextController should know how to handle surrogated pairs. Shouldn't we always pass 1 here?

For example, `adjustSelectionRectForComplexText` uses a `ComplexTextController` to advance to advance in regarding the indexes on a text run without having to handle surrogated pairs by itself.

Reply:
if surrogate pairs are handled internally by ComplexTextController, how do we determine the logical length it has advanced? the value (either 1 or 2) must be stored in SVGTextMetrics